enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,396 @@
+existing tags for cloud resources when the tag list exceeds the maximum allowed limit on cloud resource.
+
+As a cluster admin, I want to delete tags, which are not required on the cloud resources.
+
+As a cluster admin, I want to be able to ignore cloud resource from all tag operations, which helps to avoid overwriting values 
+set by cloud service provider's policies.
+
+### Goals
+
+1. Enable continuous management operations (create, append , update and delete) applicable for tagging 
+ cloud resources created by kubernetes services.
+2. Provide a new custom resource of kubernetes api for specification and status of tags.
+3. Enable continuous monitoring of tags applied and reconcile when there are changes made external to 
+the controller's API.
+4. Provide CLI for operations, cloud service provide authentication and ability to edit custom resource with ease.
+
+### Non-Goals
+
+1. Managing metadata which are not related to cloud resource tagging.
+2. Sub-set of input tags to be applied on selected cloud resource.
+3. Any conflict with cloud service provider tagging policies will not be managed by controller.
+4. Regular expression or wildcard characters for classifier tag.
+5. Metrics collection and audit logging features for controller and user operations.
+6. UI frontend for managing tags.
+
+## Proposal 
+
+A new kubernetes controller to generates actions based on tag specification defined by a custom resource. The controller
+reads existing data, applies tags based on opcode and maximum limit policy. The controller would require an initial tag or label, 
+that can be used to identify cloud resources to apply and manage tags. Cloud resources can be ignored from being managed 
+by the controller by removing the classifier tag on the cloud resource. Controller reconciles the tag list on 
+cloud resources periodically at intervals configured by sync period for the controller. 
+
+### Controller configuration
+
+#### Maximum limit policy
+
+Maximum limit policy indicates whether the input tag list can be applied partially when the length of the list exceeds the 
+allowed limit on the cloud resource. The result might differ when applied along overwrite policy. Following are the list of valid values.
+
+1. POLICY_APPLY_PARTIAL - Allows input tag list applied to be partial on cloud resource. 
+2. POLICY_APPLY_STRICT - Applies input tag list strictly in complete.
+
+### Opcodes
+
+Controller supports different opcodes for actions to add, update and delete tags. Following are the supported opcodes
+1. "add" - Controller adds tags to the cloud resources. In case of an existing entry for the tag, controller reports failure.
+2. "update" - Controller updates existing tags for the cloud resources. If there is no existing entry, controller reports failure.
+3. "delete" - Controller removes tags from the cloud resources. In case of specific errors, user intervention may be required to remove tags manually
+from the cloud resources.
```

Comment:
OCP APIs are usually declarative and describe the desired state, not how to reach it.

Reply:
Typically yes, i would have expected this to have a list of tags in the spec, then track the list of tags applied in the status. You could have an option in the spec to determine how to handle tags that are present in status but not in spec. Either you would orphan them (remove them from status without doing anything) or prune them (remove them from the cloud before removing them from status).

With operations, you then need to define a semantic for, "this operation has completed". And what does that mean if I were to re-add it to the cloud? Does the user then expect that it is re-removed? Then it's not an action any more IMO